### PR TITLE
Add gridlines for dates

### PR DESCRIPTION
**Pull request by laptou**
- Add gridlines for dates

Screenshot 2020-07-14 at 15 11 22 - Display 1

### CODE REVIEW

File: client/js/gdpr/chart/index.js [hunk 1]
```diff
@@ -246,6 +252,25 @@ function createDefs() {
...
+  grid.classList.add('grid');
+
+  for (let i = 0; i < dates.length; i++) {
+    const verticalLine = document.createElementNS(SVG_NS, 'line');
+    verticalLine.classList.add('date-line');
+
+    verticalLine.setAttribute('x1', RUN_SCALE_X * i + 'px');
+    verticalLine.setAttribute('x2', RUN_SCALE_X * i + 'px');
+    verticalLine.setAttribute('y1', '0px');
+    verticalLine.setAttribute('y2', RUN_SCALE_Y * 15 + 'px');
```
Comment (gracetian6) [inline client/js/gdpr/chart/index.js:266]: good use of magic nums (what is `15` doing here though?) 

Add constant for 15

File: client/js/gdpr/chart/index.js [hunk 2]
```diff
@@ -163,7 +167,7 @@ function createSeries(color, history) {
     series.append(marker);
     series.classList.add('series-active');
     marker.setAttribute('cx', x * RUN_SCALE_X + 'px');
-    marker.setAttribute('cy', y * RUN_SCALE_Y + 'px');
+    marker.setAttribute('cy', (y - 0.5) * RUN_SCALE_Y + 'px');
```
Comment (gracetian6) [inline client/js/gdpr/chart/index.js:170]: what's the 0.5 changes for? How are adding date lines shifting everything by 0.5?

File: client/js/gdpr/chart/index.js [hunk 3]
```diff
@@ -246,6 +252,25 @@ function createDefs() {
   return defs;
 }
 
+function createGrid(dates) {
+  const grid = document.createElementNS(SVG_NS, 'g');
+  grid.classList.add('grid');
+
+  for (let i = 0; i < dates.length; i++) {
+    const verticalLine = document.createElementNS(SVG_NS, 'line');
+    verticalLine.classList.add('date-line');
```
Comment (gracetian6) [inline client/js/gdpr/chart/index.js:261]: so this is connected with the `.chart`, `.grid`, `.date-line` in the css?

Comment (gracetian6) [reply client/js/gdpr/chart/index.js:261]: add fn comment for `createGrid()`

File: client/js/gdpr/chart/index.js (hunk 1 above)
Comment (gracetian6) [reply client/js/gdpr/chart/index.js:266]: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/line helps me understand what's going on

File: client/js/gdpr/chart/index.js (hunk 2 above)
Comment (laptou) [reply client/js/gdpr/chart/index.js:176]: this centers all of the lines vertically in the chart, whereas before they were placed too far down

Comment (geekster777) [reply client/js/gdpr/chart/index.js:175]: nit: make a variable that conveys that info or leave it in a comment:
```
const gridAlignedY = (y - 0.5) * RUN_SCALE_Y;
marker.setAttribute('cy', gridAlignedY + 'px');
...
// Y axis should be dropped half a scaled unit to vertically align it with the grid.
marker.setAttribute('cy', (y - 0.5) * RUN_SCALE_Y + 'px');
```

**Pull request merged**